indico/modules/users/util.py

```diff
@@ -421,10 +428,17 @@ def set_user_avatar(user, avatar, filename, lastmod=None):
     }
 
 
-def send_default_avatar(name=None):
-    if name:
-        text = name[0].upper()
-        color = get_color_for_username(name)
+def send_default_avatar(user):
+    """Send a user's default avatar as an SVG.
+
+       :param user: A `User` object or string (external search results, registrations)
+    """
+    if isinstance(user, str):
+        text = user[0].upper()
+        color = get_color_for_user_id(user)
+    elif user.full_name:
```

Comment:
That will fail when called via `/assets/avatar/blank.svg` since user is None

Reply:
Oh. I didn't test that edge case. When does it happen?

Reply:
anonymous users in editing timelines for example (via `_get_anonymous_user` when it's configured to not reveal the name of editors to submitters)

possibly for a pending user with no first name as well (there's a moment where you have a pending user without a user id: f90b120a135)